query/src/main/scala/filodb/query/exec/rangefn/AggrOverTimeFunctions.scala

```diff
@@ -584,4 +586,89 @@ class ChangesChunkedFunctionL extends ChangesChunkedFunction with
     }
     changes += longReader.changes(longVect, startRowNum, endRowNum)
   }
-}
\ No newline at end of file
+}
+
+abstract class QuantileOverTimeChunkedFunction(funcParams: Seq[Any], var quantileResult: Double = Double.NaN
+                                              )
+  extends ChunkedRangeFunction[TransientRow] {
+  override final def reset(): Unit = { quantileResult = Double.NaN }
+  final def apply(endTimestamp: Long, sampleToEmit: TransientRow): Unit = {
+    sampleToEmit.setValues(endTimestamp, quantileResult)
+  }
+}
+
+class QuantileOverTimeChunkedFunctionD(funcParams: Seq[Any]) extends QuantileOverTimeChunkedFunction(funcParams)
+  with ChunkedDoubleRangeFunction {
+  final def addTimeDoubleChunks(doubleVect: BinaryVector.BinaryVectorPtr,
+                                doubleReader: bv.DoubleVectorDataReader,
+                                startRowNum: Int,
+                                endRowNum: Int): Unit = {
+    require(funcParams.head.isInstanceOf[Number], "quantile parameter must be a number")
+    val q = funcParams.head.asInstanceOf[Number].doubleValue()
+    var counter = 0
+    quantileResult = if (q < 0) Double.NegativeInfinity
+    else if (q > 1) Double.PositiveInfinity
+    else {
+      var rowNum = startRowNum
+      val it = doubleReader.iterate(doubleVect, startRowNum)
+      var values = new ArrayBuffer[Double]
```

Comment:
Unfortunately `ArrayBuffer[Double]` will box the doubles, causing a Java object to be wrapped around each number, and using a ton of memory.  Instead, use `debox.Buffer` which will not box and is much more efficient.